lib/streamlit/elements/vega_charts.py

```diff
@@ -169,11 +180,19 @@ def _convert_altair_to_vega_lite_spec(altair_chart: alt.Chart) -> dict[str, Any]
     datasets = {}
 
     def id_transform(data) -> dict[str, str]:
-        """Altair data transformer that returns a fake named dataset with the
-        object id.
+        """Altair data transformer that serializes the data and
+        returns a name based on the hash of the data.
```

Comment:
nit: It also stores the serialized data in the `datasets` dict, with the generated name as the key.

Reply:
Updated the docstring